### PR TITLE
Search job

### PR DESCRIPTION
**Pull request by xx-TOOL-xx**
Added breakpoint to query and filter posted jobs. Parameters to filter:

1. Category
2. Duration
3. Duration type
4. Rate
5. English level
6. Skills

There is possibility to pass params take and skip for pagination.

### CODE REVIEW

**Review by milkavladislav — APPROVED** [review]

File: src/modules/job-post/job-post.service.ts [hunk 1]
```diff
@@ -95,6 +96,47 @@ export class JobPostService {
       .getMany();
   }
 
+  async findAndFilterAll(queryParams: searchJobFiltersDto) {
+    const { category, skills, englishLevel, duration, durationType, rate } =
+      queryParams;
+    const take = queryParams.take || 100;
```
Comment (lyudad) [inline src/modules/job-post/job-post.service.ts]: no magic number, move to constant and make appropriate name

Comment (xx-TOOL-xx) [reply src/modules/job-post/job-post.service.ts]: fixed

File: src/modules/job-post/job-post.controller.ts [hunk 2]
```diff
@@ -70,6 +72,11 @@ export class JobPostController {
     return this.jobPostService.findAll();
   }
 
+  @Get('search-job')
+  findAndFilterJobs(@Query() queryParams: searchJobFiltersDto) {
```
Comment (lyudad) [inline src/modules/job-post/job-post.controller.ts]: return type?

File: src/modules/job-post/job-post.service.ts [hunk 3]
```diff
@@ -95,6 +100,47 @@ export class JobPostService {
       .getMany();
   }
 
+  async findAndFilterAll(queryParams: searchJobFiltersDto) {
```
Comment (lyudad) [inline src/modules/job-post/job-post.service.ts]: return type for all functions

**Pull request merged**